components/Rbac/interfaces.rst

```diff
@@ -0,0 +1,141 @@
+------------------
+
+.. _component_rbac_provider_current-identity-provider-interface:
+
+CurrentIdentityProviderInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Service implementing this interface should return an instance of currently used identity.
+
+.. note::
+    For more detailed information go to `Sylius API CurrentIdentityProviderInterface`_.
+
+.. _Sylius API CurrentIdentityProviderInterface: http://api.sylius.org/Sylius/Component/Rbac/Provider/CurrentIdentityProviderInterface.html
+
+.. _component_rbac_provider_permission-provider-interface:
+
+PermissionProviderInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Service implementing this interface should return an instance of permission by given code.
+
+
+.. note::
+    For more detailed information go to `Sylius API PermissionProviderInterface`_.
+
+.. _Sylius API PermissionProviderInterface: http://api.sylius.org/Sylius/Component/Rbac/Provider/PermissionProviderInterface.html
+
+.. _component_rbac_authorization_authorization-checker-interface:
+
+AuthorizationCheckerInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Service implementing this interface should check whether current identity has permission.
+
+.. note::
+    For more detailed information go to `Sylius API AuthorizationCheckerInterface`_.
+
+.. _Sylius API AuthorizationCheckerInterface: http://api.sylius.org/Sylius/Component/Rbac/Authorization/AuthorizationCheckerInterface.html
+
+.. _component_rbac_authorization_permission-map-interface:
+
+PermissionMapInterface
+~~~~~~~~~~~~~~~~~~~~~~
+
+Service implementing this interface should return permissions by given code.
+
+.. note::
+    For more detailed information go to `Sylius API PermissionMapInterface`_.
+
+.. _Sylius API PermissionMapInterface: http://api.sylius.org/Sylius/Component/Rbac/Authorization/PermissionMapInterface.html
+
+.. _component_rbac_repository_permission-repository-interface:
+
+PermissionRepositoryInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to decouple from storage that provides permissions, you should create repository class which implements this interface.
+
+.. note::
+    For more detailed information go to `Sylius API PermissionRepositoryInterface`_.
+
+.. _Sylius API PermissionRepositoryInterface: http://api.sylius.org/Sylius/Component/Rbac/Repository/PermissionRepositoryInterface.html
+
+.. _component_rbac_repository_role-repository-interface:
+
+RoleRepositoryInterface
+~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to decouple from storage that provides roles, you should create repository class which implements this interface.
+
+.. note::
+    For more detailed information go to `Sylius API RoleRepositoryInterface`_.
+
+.. _Sylius API RoleRepositoryInterface: http://api.sylius.org/Sylius/Component/Rbac/Repository/RoleRepositoryInterface.html
+
+.. _component_rbac_resolver_permission-resolver-interface:
+
+PermissionsResolverInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Service implementing this interface should return permissions with their child permissions.
```

Comment:
I'd be a bit more specific here, cause it is critical - ``with **all** their child permissions``.